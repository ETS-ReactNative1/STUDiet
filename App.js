import React from 'react';
import { StyleSheet, Text, View, Button, SafeAreaView } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { FontAwesome } from '@expo/vector-icons';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { createStackNavigator } from '@react-navigation/stack';
import DiaryScreen from './screens/DiaryScreen';
import ProfileScreen from './screens/ProfileScreen';
import PlusScreen from './screens/PlusScreen';

const Tab = createBottomTabNavigator();
const RootStack = createStackNavigator();

export default function App() {
  return (
    <SafeAreaView style={{ flex: 1 }}>
      <NavigationContainer>
        <RootStack.Navigator mode="modal" headerMode="none">
          <RootStack.Screen name="Main" component={MainStackContainer} />
          <RootStack.Screen name="PlusModal" component={PlusScreen} />
        </RootStack.Navigator>
      </NavigationContainer>
    </SafeAreaView>
  );
}

function MainStackContainer({ navigation }) {
  return (
    <Tab.Navigator
      screenOptions={({ route }) => ({
        tabBarIcon: ({ focused, color, size }) => {
          let iconName;

          if (route.name === 'Diary') {
            iconName = 'list-alt';
          } else if (route.name === 'Profile') {
            iconName = 'user';
          }

          return <FontAwesome name={iconName} size={size} color={color} />;
        },
      })}
      tabBarOptions={{
        activeTintColor: 'red',
        inactiveTintColor: 'gray',
      }}
      initialRouteName="Diary"
    >
      <Tab.Screen name="Diary" component={DiaryScreen} />
      <Tab.Screen
        name="Plus"
        component={PlusScreen}
        options={{
          tabBarButton: () => (
            <FontAwesome
              onPress={() => navigation.navigate('PlusModal')}
              name="plus-circle"
              size={50}
            />
          ),
        }}
      />
      <Tab.Screen name="Profile" component={ProfileScreen} />
    </Tab.Navigator>
  );
}
