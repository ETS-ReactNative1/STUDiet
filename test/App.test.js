import React from 'react'
import renderer, { act } from 'react-test-renderer'

import App from '../App'
import AppContainer from '../navigation'
describe('<App />', () => {
    it('renders correctly', () => {
        act(() => {
            const tree = renderer.create(<App />).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})
describe('<AppContainer />', () => {
    it('renders correctly', () => {
        act(() => {
            const tree = renderer.create(<AppContainer />).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})
const tree = renderer.create(<AppContainer />).toJSON()
console.log(tree)
