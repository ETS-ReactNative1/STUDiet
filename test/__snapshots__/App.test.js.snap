// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `null`;

exports[`<AppContainer /> renders correctly 1`] = `null`;
